### PR TITLE
Fixes #38564: do not check kdump-options with .presence

### PR DESCRIPTION
**Pull request by d1nuc0m**
* Check if the variable exists and is not empty
* Add %end to close addon section

### CODE REVIEW

File: app/views/unattended/provisioning_templates/provision/kickstart_default.erb [hunk 1]
```diff
@@ -140,10 +140,11 @@ firewall --disable
 firewall --<%= os_major >= 6 ? 'service=' : '' %>ssh
 <% end -%>
 <%# Kdump -%>
-<%# Ignore empty strings, & because .presence isn't allowed on nil
-See https://github.com/theforeman/foreman/pull/10507 -%>
-<% if host_param('kdump-options')&.presence -%>
+<%# .presence seem not to work in this case, instead check if the variable exists and is not empty
+See https://projects.theforeman.org/issues/38564 -%>
+<% if host_param('kdump-options') && !host_param('kdump-options').empty? -%>
```
Comment (ekohl) [inline app/views/unattended/provisioning_templates/provision/kickstart_default.erb]: I think this should actually work safely even within a jail.
```suggestion
<% if host_param('kdump-options').present? -%>
```

**Review by ekohl — CHANGES_REQUESTED** [review]
Perhaps it would be a good idea to add the parameter to the tests:
https://github.com/theforeman/foreman/blob/ef417a172b3681e5670d3e268f1e7bc3178936eb/app/services/foreman/template_snapshot_service.rb#L68-L101

That way we add test coverage. We also run [pykickstart](https://pykickstart.readthedocs.io/) on it to validate the result.

File: app/views/unattended/provisioning_templates/provision/kickstart_default.erb (hunk 1 above)
Comment (d1nuc0m) [reply app/views/unattended/provisioning_templates/provision/kickstart_default.erb]: What would happen if it is present but nil?

**Comment by d1nuc0m** [comment]
> Perhaps it would be a good idea to add the parameter to the tests:
> 
> https://github.com/theforeman/foreman/blob/ef417a172b3681e5670d3e268f1e7bc3178936eb/app/services/foreman/template_snapshot_service.rb#L68-L101
> 
> That way we add test coverage. We also run [pykickstart](https://pykickstart.readthedocs.io/) on it to validate the result.

Is it enough to add it to host_params?

File: app/views/unattended/provisioning_templates/provision/kickstart_default.erb (hunk 1 above)
Comment (ekohl) [reply app/views/unattended/provisioning_templates/provision/kickstart_default.erb]: https://blog.appsignal.com/2018/09/11/differences-between-nil-empty-blank-and-present.html#present says `nil.present?` returns `false`.

**Comment by ekohl** [comment]
> Is it enough to add it to host_params?

That, but I'd expect that you also need to update the snapshots that will now have the addon text.

**Comment by d1nuc0m** [comment]
> > Is it enough to add it to host_params?
> 
> That, but I'd expect that you also need to update the snapshots that will now have the addon text.

Right, is it ok to always have the parameter enabled or should also we test nil and no param cases?

File: app/views/unattended/provisioning_templates/provision/kickstart_default.erb [hunk 2]
```diff
@@ -140,10 +140,11 @@ firewall --disable
 firewall --<%= os_major >= 6 ? 'service=' : '' %>ssh
 <% end -%>
 <%# Kdump -%>
-<%# Ignore empty strings, & because .presence isn't allowed on nil
-See https://github.com/theforeman/foreman/pull/10507 -%>
-<% if host_param('kdump-options')&.presence -%>
+<%# .presence seem not to work in a jail
+See https://github.com/theforeman/foreman/pull/10605 -%>
```
Comment (ekohl) [inline app/views/unattended/provisioning_templates/provision/kickstart_default.erb]: I think these kind of comments just muddy the whole template. If anything, I'd leave them in the git commit message.
```suggestion
```

**Review by ekohl — COMMENTED** [review]
Minor note, but otherwise 👍

File: app/views/unattended/provisioning_templates/provision/kickstart_default.erb (hunk 2 above)
Comment (d1nuc0m) [reply app/views/unattended/provisioning_templates/provision/kickstart_default.erb]: Done, can't understand why some tests are failing though

**Review by ekohl — APPROVED** [review]
Thanks. I verified with the RH docs that at least since RHEL 7 the `%end` is mandatory. For kdump there don't appear to be options, but with OpenSCAP it's used to provide those.

**Comment by ekohl** [comment]
CI is still running, but should be good to be merged.

**Comment by ekohl** [comment]
Katello changes look unrelated. Thanks!
